# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1319665014 14400
#      Wed Oct 26 17:36:54 2011 -0400
# Node ID 1e81e2e30af33f619976dc8b8af49417bba860b1
# Parent  6dfebfa334cbc63191cdb946e16209eb23852443
default limits for logscale plots

* graphics.cc (default_lim): New optional argument, logscale.
Return [0.1, 1.0] if logscale is true.
(axes::properties::get_axis_limits): Return default limits if min/max
values are at their initial values.

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -215,20 +215,28 @@ default_colororder (void)
   retval(6,0) = 0.25;
   retval(6,1) = 0.25;
   retval(6,2) = 0.25;
 
   return retval;
 }
 
 static Matrix
-default_lim (void)
+default_lim (bool logscale = false)
 {
   Matrix m (1, 2, 0);
-  m(1) = 1;
+
+  if (logscale)
+    {
+      m(0) = 0.1;
+      m(1) = 1.0;
+    }
+  else
+    m(1) = 1;
+
   return m;
 }
 
 static Matrix
 default_data (void)
 {
   Matrix retval (1, 2);
 
@@ -5549,17 +5557,22 @@ axes::properties::get_axis_limits (doubl
                                    double min_pos, double max_neg,
                                    bool logscale)
 {
   Matrix retval;
 
   double min_val = xmin;
   double max_val = xmax;
 
-  if (! (xisinf (min_val) || xisinf (max_val)))
+  if (xisinf (min_val) && min_val > 0 && xisinf (max_val) && max_val < 0)
+    {
+      retval = default_lim (logscale);
+      return retval;
+    }
+  else if (! (xisinf (min_val) || xisinf (max_val)))
     {
       if (logscale)
         {
           if (xisinf (min_pos) && xisinf (max_neg))
             {
               // TODO -- max_neg is needed for "loglog ([0 -Inf])"
               //         This is the only place where max_neg is needed.
               //         Is there another way?
